src/test/java/picard/illumina/ExtractIlluminaBarcodesTest.java

```diff
@@ -391,4 +395,48 @@ private void testParsing(final IlluminaDataProviderFactory factory, final ReadSt
         retval.read(new FileReader(metricsFile));
         return retval;
     }
+
+    @DataProvider
+    Object[][] testDeltaData(){
+        return new Object[][]{
+                new Object[] {new String[]{"ACCAAC", "GAATTC"}, new String [] {"!AAAAA","!AAAAA"}, 2, 5},
+                new Object[] {new String[]{"CTACGC", "TGTCGT"}, new String [] {"!AAAAA","!AAAAA"}, 5, 5},
+                new Object[] {new String[]{"AGGTCG", "AATTGT"}, new String [] {"AAAAAA","AAAAAA"}, 0, 5},
+        };
+    }
+
+    @Test(dataProvider = "testDeltaData")
+    void testDelta(final String[] barcodeRead, final String[] barcodeQuality, final int expectedMismatches, final int expectedSecondMismatches) {
+        Map<String, BarcodeMatch> barcodeMatches = new HashMap<>();
+
+        List<String[]> barcodes = Arrays.asList(
+                new String[]{"CTGTGG", "GGCTAG"},
+                new String[]{"AGGTCG", "AATTGT"},
+                new String[]{"ACCAAC", "GTATTG"}
+                );
+        Map<String, ExtractIlluminaBarcodes.BarcodeMetric> barcodeMetrics = barcodes.stream()
+                .collect(Collectors.toMap(
+                        s -> s[0] + s[1],
+                        s -> new ExtractIlluminaBarcodes.BarcodeMetric(s[0] + s[1], "dummy", "more_dummy", s)));
+
+        byte[][] reads = new byte[][]{
+                barcodeRead[0].getBytes(),
+                barcodeRead[1].getBytes()
+        };
+        byte[][] qualities = new byte[][]{
+                barcodeQuality[0].getBytes(),
+                barcodeQuality[1].getBytes()
+        };
+        BarcodeMatch match = ExtractIlluminaBarcodes.PerTileBarcodeExtractor.findBestBarcode(reads, qualities,
+                barcodeMetrics, 2, 2, 2, 20,
+                barcodeMatches, DistanceMetric.HAMMING);
+
+
+        Assert.assertFalse(barcodeMatches.isEmpty());
+        final BarcodeMatch barcodeMatch = barcodeMatches.get(String.join("-",barcodeRead[0], barcodeRead[1]));
```

Comment:
Unless you are trying to explicitly test that `barcodeMatches` has been correctly updated, you can just use `match` in all the assertions.

Reply:
Also if you move the lookup table modifications out of `findBestBarcode`, then will definitely need to just use `match`.